Network/HTTP/ReverseProxy.hs

```diff
@@ -75,7 +77,12 @@ import           UnliftIO                       (MonadIO, liftIO, MonadUnliftIO,
 data ProxyDest = ProxyDest
     { pdHost :: !ByteString
     , pdPort :: !Int
-    } deriving (Read, Show, Eq, Ord, Generic)
+    }
+    | ProxyDestUnix {
```

Comment:
I'm not a fan of this approach, since it creates partial functions as well as breaking existing code that pattern matches. A new data type would be necessary.

Reply:
Okay, could you clarify what you have in mind? Avoiding a breaking change would seem to require a lot of duplication. For example, where we have

```haskell
rawTcpProxyTo :: MonadIO m => ProxyDest -> AppData -> m ()
```

I could create a parallel universe of functions like 

```haskell
rawTcpProxyToUnix :: MonadIO m => ProxyDestUnix -> AppData -> m ()
```

but that would require duplicating most of public proxying functions as well as `WaiProxyResponse`...

Reply:
Unfortunately I don't have a specific recommendation in mind.

Reply:
Would you accept a full breaking change, renaming the data type and bumping the version number?

Reply:
I would lean towards creating a parallel set of functions.